src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -425,6 +440,38 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+/// <summary>
+/// Set the CustomHttpHeaders property from the custom http headers string.
+/// </summary>
+/// <param name="stringValue">The value to set.</param>
+public void ValidateAndSetCustomHttpHeadersFromString(string headersValue) 
+{
+    if (String.IsNullOrWhiteSpace(headersValue))
+    {
+        return;
+    }
+
+    this.CustomHttpHeaders = new List<string>();
+
+    string[] headerElements = headersValue.Split(new string[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);

```

Comment:
```suggestion
    string[] headerElements = headersValue.Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries);
```

Any reason for passing the separator as as `string[]` instead of just `string`?

Reply:
The split method only receives char[] and string[]
https://docs.microsoft.com/en-us/dotnet/api/system.string.split?view=netframework-4.8

Reply:
Interesting. Seems only .netframework doesn't have it as an option. Core and standard both have it